### DISCUSSION TITLE
ghcr support

### DISCUSSION THREAD

**Discussion by Zoey2936** [nextcloud/all-in-one#2853] [Ideas] [answered]
If AIO supports ghcr, PRs could be build and pushed to ghcr and with that they could be easily tested (like I do with my repos: https://github.com/ZoeyVid/nginx-proxy-manager/pull/313#issuecomment-1602973853)

**Comment by szaimen** [comment]
I think testing is already easy enough: https://github.com/nextcloud/all-in-one/blob/main/develop.md#developer-channel or what would be the advantages?

**Reply by Zoey2936** [reply]
it can be tested before beeing merged

**Reply by szaimen** [reply]
Hm... not sure about this idea. It sounds like making this possible would be a big effort due to AIO architecture and since I would in the end still have to test the devlop tag on docker hub (in order to be as close to production as possible and also to test everything together), it sounds like this would double the testing time... 

However what could be useful imho would be adding a workflow that runs docker buld whenever things are pushed to the specific container folder. what do you think about this idea?

**Reply by Zoey2936** [reply]
so that the images could then only be tested for their own and not by running for example `docker run ghcr.io/nextcloud/all-in-one:pr-2438`, but `docker run ghcr.io/nextcloud/aio-nextcloud:pr-2438`

**Reply by szaimen** [reply]
but would that actually be useful? I mean you would still need to extract the configs from containers.json and the run it with their correct config and some containers need others present to even start...

**Reply by Zoey2936** [reply, marked as answer]
then lets leave it as it is
